src/sentry/api/endpoints/event_owners.py

```diff
@@ -35,11 +35,26 @@ def get(self, request, project, event_id):
         if owners == ProjectOwnership.Everyone:
             owners = []
 
+        serialized_owners = serialize(Actor.resolve_many(owners), request.user, ActorSerializer())
+        # We do so many dict/set casts on these owners that the order is not preserved at all.
+        # Re-order the results according to how the rules are ordered.
+        owner_map = {o["name"]: [o, False] for o in serialized_owners}
+        ordered_owners = []
+        for rule in rules:
+            for o in rule.owners:
+                found = owner_map.get(o.identifier)
+                if found:
+                    ordered_owners.append(found[0])
+                    found[1] = True
+
+        # A litle defensive, just in case there are owners that aren't part of the rules, add them to the end.
+        for name in owner_map:
+            if not owner_map[name][1]:
```

Comment:
Should this loop only be run if `len(serialized_owners) != len(ordered_owners)`?

Reply:
Yeah if this is useful I'll change it to make that check.